hudi-common/src/main/java/org/apache/hudi/common/engine/HoodieReaderContext.java

```diff
@@ -87,6 +89,8 @@ public abstract class HoodieReaderContext<T> {
   private RecordMergeMode mergeMode;
   protected RecordContext<T> recordContext;
   private FileGroupReaderSchemaHandler<T> schemaHandler = null;
+  // the default iterator mode is engine-specific record mode
+  private IteratorMode iteratorMode = IteratorMode.ENGINE_RECORD;
```

Comment:
I don't think this makes a lot of sense as a member of the ReaderContext, is it possible to keep it confined to the FileGroupReader and RecordBuffer classes?